syntax = "proto3";

package serialize_transport_catalogue;

message RoutingSettings {
    double bus_wait_time = 1;
    double bus_velocity = 2;
}

message VertexInfo {
    string stop_name = 1;
    bool is_bus_stop = 2;
    uint32 id = 3;
}

message EdgeInfo {
    bool is_bus_edge = 1;
    uint32 from_id = 2;
    uint32 to_id = 3;
    int32 span_count = 4;
    string name = 5;
    double weight = 6;
}

message GraphInfo {
    repeated VertexInfo vertexes = 1;
    repeated EdgeInfo edges = 2;
    RoutingSettings routing_settings = 3;
}